src/mscorlib/shared/System/BitConverter.cs

```diff
@@ -21,6 +21,20 @@ public static class BitConverter
 #else
         public static readonly bool IsLittleEndian = true;
 #endif
+        private static T ReadUnaligned<T>(ReadOnlySpan<byte> value) where T : struct
+        {
+            if (value.Length < Unsafe.SizeOf<T>())
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.value);
+            return Unsafe.ReadUnaligned<T>(ref value.DangerousGetPinnableReference());
+        }
+
+        private static bool WriteUnaligned<T>(T value, Span<byte> bytes) where T : struct
+        {
+            if (bytes.Length < Unsafe.SizeOf<T>())
```

Comment:
> generics have overhead

Is the overhead prohibitive in this case?  I'd suggested this approach to @nchikanov as a way of avoiding lots of duplicated code, and I assumed that since all of the Ts were primitives/structs, it would essentially end up being equivalent to manually inlining the implementation.  But it sounds like that's not the case?